### PR TITLE
feat: add Euler Angle class

### PR DESCRIPTION
**Pull request by lucas-bremond**
Add Euler Angle class, with only support for `XYZ`, `ZXY` and `ZYX` sequences.

### CODE REVIEW

**Review by Derollez — APPROVED** [review]
Amazing! Let us know how the csv were generated as source of truth and we can add the other sequences if necessary :)

**Review by kyle-cochran — APPROVED** [review]
Small stuff

File: include/OpenSpaceToolkit/Mathematics/Geometry/3D/Transformation/Rotation/EulerAngle.hpp [hunk 1]
```diff
@@ -16,7 +23,350 @@ namespace transformation
...
+
+    /// @brief                  Get first angle of rotation (phi)
+    ///
+    /// @code
+    ///                         EulerAngle::Unit().phi();
+    /// @endcode
+    ///
+    /// @return                 First angle of rotation
+
+    Angle phi() const;
```
[the remark this answers is not in the document]
Comment (kyle-cochran) [reply include/OpenSpaceToolkit/Mathematics/Geometry/3D/Transformation/Rotation/EulerAngle.hpp:175]: Nevermind, this is consistent with quaternions at least. Resolving

File: include/OpenSpaceToolkit/Mathematics/Geometry/3D/Transformation/Rotation/EulerAngle.hpp [hunk 2]
```diff
@@ -16,7 +23,350 @@ namespace transformation
...
+    /// @code
+    ///                         EulerAngle::Unit().toVector(Angle::Unit::Degree);
+    /// @endcode
+    ///
+    /// @param                  [in] anAngleUnit An angle unit
+    /// @return                 Vector containing the three angles
+
+    Vector3d toVector(const Angle::Unit& anAngleUnit) const;
+
+    /// @brief                  Return string from of Euler Angle [phi; theta; psi]
```
[the remark this answers is not in the document]
Comment (lucas-bremond) [reply include/OpenSpaceToolkit/Mathematics/Geometry/3D/Transformation/Rotation/EulerAngle.hpp]: Done. Thanks!

File: include/OpenSpaceToolkit/Mathematics/Geometry/3D/Transformation/Rotation/Quaternion.hpp [hunk 3]
```diff
@@ -77,527 +78,515 @@ class Quaternion
...
+    ///                             Angle::Radians(1e-6)
+    ///                         );
     /// @endcode
     ///
-    /// @param              [in] aQuaternion A quaternion
-    /// @param              [in] anAngularTolerance An angular tolerance
-    /// @return             True if quaternions are near each other
+    /// @param                  [in] aQuaternion A Quaternion
+    /// @param                  [in] anAngularTolerance An angular tolerance
+    /// @return                 True if Quaternion are near each other
```
[the remark this answers is not in the document]
Comment (lucas-bremond) [reply include/OpenSpaceToolkit/Mathematics/Geometry/3D/Transformation/Rotation/Quaternion.hpp]: Done. Thanks!

**Pull request merged**